contracts/multisig/src/signing.rs

```diff
@@ -183,111 +153,75 @@ mod tests {
-
-        sign(
-            &mut config.store,
-            session,
             key,
-            signer.address.into_string(),
-            Signature::try_from((config.key_type, signer.signature.clone())).unwrap(),
-        )
+            session,
+            signatures,
+            key_type,
+        }
     }
 
     #[test]
-    fn test_add_signature() {
-        for mut config in [ecdsa_setup(), ed25519_setup()] {
-            let mut session =
-                SigningSession::new(Uint64::one(), config.key_id.clone(), config.message.clone());
-
-            // first run
-            let result = do_sign(&session, 0, &mut config);
-            let stored = SIGNING_SESSIONS
-                .load(&config.store, session.id.u64())
-                .unwrap();
-            let signatures = session_signatures(&config.store, session.id.u64()).unwrap();
-
-            assert!(result.is_ok());
-            assert_eq!(signatures.len(), 1);
-            assert_eq!(stored, session);
+    fn test_correct_session_state() {
+        for config in [ecdsa_setup(), ed25519_setup()] {
+            let mut session = config.session;
+            let key = config.key;
+            let signatures = config.signatures;
+            let block_height = 12345;
 
-            // second run
-            let result = do_sign(&mut session, 0, &mut config);
-            let stored = SIGNING_SESSIONS
-                .load(&config.store, session.id.u64())
-                .unwrap();
-            let signatures = session_signatures(&config.store, session.id.u64()).unwrap();
+            session.recalculate_session_state(&HashMap::new(), &key.snapshot, block_height);
+            assert_eq!(session.state, MultisigState::Pending);
 
+            session.recalculate_session_state(&signatures, &key.snapshot, block_height);
             assert_eq!(
-                result.unwrap_err(),
-                ContractError::DuplicateSignature {
-                    session_id: session.id,
-                    signer: config.signers[0].address.clone().into_string()
+                session.state,
+                MultisigState::Completed {
+                    completed_at: block_height
                 }
             );
-            assert_eq!(signatures.len(), 1);
-            assert_eq!(stored, session);
         }
     }
 
     #[test]
-    fn test_calculate_session_state() {
-        for mut config in [ecdsa_setup(), ed25519_setup()] {
-            let session =
-                SigningSession::new(Uint64::one(), config.key_id.clone(), config.message.clone());
-
-            // first run
-            let (_, result) = do_sign(&session, 0, &mut config).unwrap();
-            assert_eq!(result, MultisigState::Pending);
-
-            // second run
-            let (_, result) = do_sign(&session, 1, &mut config).unwrap();
-            assert_eq!(result, MultisigState::Completed);
+    fn test_success_validation() {
```

Comment:
no need to start test functions with `test_`, they are already clearly denoted by the attribute and general rust convention forgoes the prefix